src/components/annotations/attributes.js

```diff
@@ -15,7 +15,7 @@ var extendFlat = require('../../lib/extend').extendFlat;
 
 
 module.exports = {
-    _isLinkedToArray: true,
+    _isLinkedToArray: 'annotation',
```

Comment:
> I've never been able to figure out quite what _isLinkedToArray

It's essentially the plot schema version of `Plots.layoutArrayContainers`. With this PR, both are now united.

Giving a name to the items in the array container is (again) only relevant for `plotly.py` at the moment. Previously, the item names were simply the container name minus the `'s`'. This PR make item naming more robust / flexible.